packages/react-client/src/__tests__/ReactFlight-test.js

```diff
@@ -302,4 +304,220 @@ describe('ReactFlight', () => {
       {withoutStack: true},
     );
   });
+
+  it('supports basic createServerContext usage', () => {
+    const ServerContext = React.createServerContext(
+      'ServerContext',
+      'hello from server',
+    );
+    function Foo() {
+      const context = React.useServerContext(ServerContext);
+      return <div>{context}</div>;
+    }
+
+    const transport = ReactNoopFlightServer.render(<Foo />);
+    act(() => {
+      ServerContext._currentRenderer = null;
```

Comment:
I tried using `jest.isolateModules` so that the instances of `ReactServerContext` on the server/client were different but that ended up breaking the previous tests for some reason and didn't fix my issue. I can look into a better way to test this but for now is probably sufficient?

Reply:
It does surface an interesting scenario. Shopify currently runs Flight and Fizz in the same environment where they could interleave with each other. We don't really support it yet but there's a discussion about whether we should. Perhaps we should flip the isPrimaryRenderer bit for Flight so that if you end up in that situation you could at least make sure one renderer works with both. That might also fix your tests.